### DISCUSSION TITLE
DA18x20 temperature sensor, what do i need to select in that list?

### DISCUSSION THREAD

**Discussion by jayjupdhig** [arendst/Tasmota#12405] [Q&A] [answered]
Hi there

I mean that one, we connected it with a cable similar to an audio cable on a Sonoff TH16: https://www.google.com/search?q=ds18x20&tbm=isch&ved=2ahUKEwivvqv2gKHxAhVcgv0HHSBoDLcQ2-cCegQIABAA&oq=ds18x20&gs_lcp=CgNpbWcQDFAAWABgxqgIaABwAHgAgAEAiAEAkgEAmAEAqgELZ3dzLXdpei1pbWc&sclient=img&ei=4nnMYK_FCNyE9u8PoNCxuAs&bih=969&biw=1920

![Unbenannt](https://user-images.githubusercontent.com/42765391/122550785-649fc080-d034-11eb-8192-22cd9a93e7b5.jpg)

Thank you very very much for your feedback. Best regards, Jan

**Comment by sfromis** [comment]
What binary are you using there? Looks like a cutdown one. Try upgrading to full, tasmota.bin.gz, and you should find `DS18x20` in the dropdown.

**Comment by lalo-uy** [comment]
TH16 put the sensor on GPIO14

**Comment by jayjupdhig** [comment, marked as answer]
@sfromis: Thank you!!

@lalo-uy: That's what i have done... that connection (similar to audio cable on sonoff th) is not "Plug and Play", no, not really... ;-)

**Reply by barbudor** [reply]
@jayjupdhig Did you purchased a DS18x20 probe specifically made to go with TH16 ? Or did you made it yourself ?
If you purchased a random DS18x20 probe with a jack connector, it may not be the right pinout for the TH16. As far as I know, Sonoff do not sell DS18x20 for TH16, so even if Tasmota can handle this, if your hardware connection is not correct, it won't work and you may even fry your DS18x20

**Reply by jayjupdhig** [reply]
Our engineers made it (DS18x20) work with Sonoff TH16, but that is not the problem so it seems. The problem is much more the "lite" version - otherwise the DS18x20-entry would appear in the list...

**Reply by barbudor** [reply]
Indeed, the lite version does include DS18x20
You need at least tasmota.bin(.gz)

**Reply by sfromis** [reply]
Stricly speaking  iTead does have the DS18B20 sensor as an option when buying Sonoff TH10/TH16, but that may just be their own "standard" for how the plug bands are used.

**Reply by jayjupdhig** [reply]
How can i update the 9.1 LITE to 9.1 FULL? When i use the .bin file, i get that buffer error message: https://tasmota.github.io/docs/Upgrading/

And, so i saw, we can only reach lite version installation with that step-by-step upgrade guide: "v1.0.11  🔀  v3.9.22  🔀  v4.2.0  🔀  v5.14.0  🔀  v6.7.1  🔀  v7.2.0  🔀  v8.5.1  🔀  v9.1  🔀  Current release"

v3.9.22 -> links to a file "firmware.bin"

v4.2.0 -> links to a file "firmware.bin"

 v5.14.0 -> links to a file "sonoff-classic.bin"

 v6.7.1 -> links to a file "sonoff.bin"

 v7.2.0 -> links to a file "tasmota-lite.bin" (??????)
 
v8.5.1  -> links to a file "tasmota-lite.bin" (??????)

v9.1 -> -> links to a file "tasmota-lite.bin.gz" (??????)


That's the only one step-by-step update process - with lite files. So i know, our support employee has done this. And the result is that what i have - a missing DS18x20 entry...

QUESTION: How to solve my problem (9.1 LITE to 91. FULL)as fast as possible? Maybe a fresh flashing (not web-ui based updating) via RS-232?

Or was it wrong to use the 9.1 FULL .bin file instead of 9.1 FULL .bin.gz file??

Otherwise, i need to use the soldering iron... ;-)

**Reply by barbudor** [reply]
To upgrade file-upload , use the bin.gz file
bin only files are for serial flashing

If your device has access to the Internet, you can also upgrade using OTA using one of the below URL:
Release 9.5.0 : http://ota.tasmota.com/tasmota/release/tasmota.bin.gz
Release 9.1.0 : http://ota.tasmota.com/tasmota/release-9.1.0/tasmota.bin.gz
Development 9.5.0.1 : http://ota.tasmota.com/tasmota/tasmota.bin.gz

**Reply by sfromis** [reply]
You can upgrade to another binary, base, lite or sensors anytime, like going from lite to full. The lite binaries are great to use during the stepwise upgrading process thru versions, takes fewer steps, and is thus faster. When you are at the current release, next step is to upgrade to full.

You could also use non-lite versions earlier, but then each step takes first using minimal and then full (both for each step). Using lite, you can generally avoid those extra minimal steps.

**Comment by jayjupdhig** [comment]
Got it - works!! Thank you very much!!!
